src/Domain/Common/AuditableEntity.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Domain.Common;
+
+public abstract class AuditableEntity : Validable
```

Comment:
Do we only care about the last change? What about history of changes? Perhaps implementing a system-wide notification system instead of method in each class would be more scalable? A notification could be handled by whoever is interested, e.g. a history tracker, or a property updater (instead of forcing each class to update the properties itself). This is a difficult and deep topic I believe. What are the arguments for implementing these properties in the code to begin with?

Reply:
You're right, but I think this approach can be a compromise. It offers some kind of minimun amount of auditability while keeping it simple. 
To get a fully audit system then "event sourcing" could be the perfect solution.
Could be fun to implement an ES version of this template, wouldn't it? :-)